### ISSUE TITLE
[VSCode] The Tabby chat panel text is not visible in dark theme mode.

### ISSUE THREAD

**Issue by Sma1lboy** [TabbyML/tabby#3031]
**Describe the bug**

The text is generally black and gray, and even with the default VSCode theme, it is difficult to see clearly. I'm not sure if I set it up incorrectly.

<img width="399" alt="image" src="https://github.com/user-attachments/assets/a01b4046-614d-4a5b-9965-4c3555485367">

**Comment by wsxiaoys** [comment]
Related discussion: https://tabbyml.slack.com/archives/C05CWLZ0Y85/p1724338919612139

The issue is likely due to the function `syncTheme` not handling some edge cases properly: https://github.com/TabbyML/tabby/blob/2852d3a09fa412ff6f288c6646d04bbb1541f73b/clients/vscode/src/chat/ChatViewProvider.ts#L363

Currently, I don't have a way to reproduce the issue, but if you are able to do so, you can inspect the VS Code window by using the command `Open Webview Developer Tools` to see if you can locate the problematic logic.

**Comment by gdmka** [comment]
@wsxiaoys which version of VSC are you running?
I do feel this may be of importance.  When I asked for assistance on Slack and to this day mine is 
`Version: 1.83.1
Commit: f1b07bd25dfad64b0167beb15359ae573aecd2cc`

**Comment by wsxiaoys** [comment]
<img width="229" alt="image" src="https://github.com/user-attachments/assets/1914657d-d303-489a-80b3-82b4b3a2b4fb">

So it's probably an 1.8x version issue?

**Comment by Sma1lboy** [comment]
This is my vscode version

```
❯ code --version
1.92.2
fee1edb8d6d72a0ddff41e5f71a671c23ed924b9
arm64

```

**Comment by gdmka** [comment]
Alas, the issue persists for me with the latest stable VSC build. 

![image](https://github.com/user-attachments/assets/780cd4eb-eab8-4fee-b613-d61228f28e13)

A good idea is to have some dynamic adaptation for [dark mode designs](https://github.com/tailwindlabs/tailwindcss-typography?tab=readme-ov-file#adapting-to-dark-mode) that may fit everyone. 
Where these attributes are set explicitly:

```
--tw-prose-body:  #ffffff;
--tw-prose-code: #2b6fe3; // this one suits me doesn't have to be explicitly this color
```

I have attached a small demo @wsxiaoys 

![Sep-05-2024 16-26-03](https://github.com/user-attachments/assets/df22253e-96da-4dbd-b04b-f76254340916)

**Comment by wsxiaoys** [comment]
@gdmka can you share the change you made to make font color works for you?

**Comment by gdmka** [comment]
Sure @wsxiaoys 

I've recorded step by step what I do to make the TabbyML sidebar work for me. Given the changes are temporary, I'd have to do it every time I launch VSC.

First I adjust the color of the general text. 

https://github.com/user-attachments/assets/5e68974a-b926-4794-b7d2-940be9dfc3d2

Then I adjust the symbol name CSS value. 


https://github.com/user-attachments/assets/fbf0b47f-bb9e-444a-8cd3-9bd2b9e236cb


And after adjustments, consecutive requests end in user friendly rendering of text and symbol names. 


https://github.com/user-attachments/assets/9169e4d7-9427-4b35-8665-7170a5db9a82


I've listed acceptable CSS values for Tailwind vars in my previous [message](https://github.com/TabbyML/tabby/issues/3031#issuecomment-2331834331) just in case.

**Comment by wsxiaoys** [comment]
We've identified the cause and working on a fix, it will be released in 1.10.3

**Comment by gdmka** [comment]
> We've identified the cause and working on a fix, it will be released in 1.10.3

Is it deeper than what was described before?

**Comment by wsxiaoys** [comment]
The fix can be found at https://github.com/TabbyML/tabby/pull/3106/files. The issue was that, in certain environments, `document.body.className` contained more than just the `vscode-dark` theme string. It sometimes included other strings, such as `vscode-reduce-motion`, which caused the `getTheme` detection to fail.

1.10.3 Released: https://marketplace.visualstudio.com/manage/publishers/tabbyml/extensions/vscode-tabby/hub?_a=manageExtension
